crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1515,12 +1513,28 @@ where
             move |name, client| {
                 Box::pin(async move {
                     let _guard = GaugeGuard::acquire(&metrics_clone.inflight_certificate_requests);
-                    client
-                        .handle_certificate_v2(cert_ref, client_addr)
-                        .instrument(
-                            tracing::trace_span!("handle_certificate", authority =? name.concise()),
-                        )
-                        .await
+                    if request_ref.include_input_objects || request_ref.include_output_objects {
+                        client
+                            .handle_certificate_v3(request_ref, client_addr)
```

Comment:
Two questions here:
1. I think we need to do some basic sampling to avoid asking every validator for this.
2. Why do we need this if/else? We could just always use the v3 path?

Reply:
> I think we need to do some basic sampling to avoid asking every validator for this.

Sampling has been added.

> Why do we need this if/else? We could just always use the v3 path?

Yes we could and I plan to remove the v2 path eventually (say another release or 2), but given this is a new api that is being rolled out i figured it would be best to have all requests that dont need the new functionality to continue using the old v2 path until validators have actually been updated and running the new v3 path, otherwise we could get into a situation where someone updates a fullnode and it tries to start using the new path before the validators had updated.